<robot name="test_robot">
  <link name="base_link">
        <visual>
                <geometry>
                        <sphere radius="0.3"/>
                </geometry>
		<material name = "blue">
			<color rgba = "0 0 1 1"/>
		</material>
        </visual>
  </link>

  <link name="arm1">
	<visual>
		<geometry>
			<cylinder length="2" radius="0.05"/>
		</geometry>
		<origin xyz = "0 0 1"/>
		<material name = "blue"/>
	</visual>
  </link>
    
  <link name="link1">
	<visual>
		<geometry>
			<sphere radius="0.15"/>
		</geometry>
		<material name = "blue"/>
		<origin xyz = "0 0 2" rpy="0 1.57 0"/>
	</visual>
  </link>

  <joint name="link1_to_arm1" type="fixed">
    <parent link = "link1"/>
    <child link = "arm1"/>
  </joint>

  <link name="arm2">
        <visual>
                <geometry>
                        <cylinder length= "1" radius="0.05"/>
                </geometry>
		<material name = "blue"/>
                <origin xyz = "0.5 0 0" rpy="0 1.57 0"/>
        </visual>
  </link>


  <link name="link2">
        <visual>
                <geometry>
                        <sphere radius="0.15"/>
                </geometry>
		<material name = "blue"/>
	<origin xyz = "1 0 0"/>

        </visual>
  </link>

  <joint name="link2_to_arm2" type="fixed">
    <parent link = "link2"/>
    <child link = "arm2"/>
  </joint>

  <link name="arm3">
        <visual>
                <geometry>
                        <cylinder length="3" radius="0.025"/>
                </geometry>
                <origin xyz = "0 0 -1.5"/>
		<material name = "blue"/>
        </visual>
  </link>


  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0" rpy="0 1.57 0" />
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <axis xyz="1 0 0" />
  </joint>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 2" rpy="0 -1.57 0" />
    <limit effort="1000.0" lower="-2.35619" upper="2.35619" velocity="0.5"/>
    <axis xyz="0 0 1" />
  </joint>

  <joint name="joint3" type="prismatic">
    <parent link="link2"/>
    <child link="arm3"/>
    <limit effort="1000.0" lower="0" upper="3" velocity="0.5"/>
    <origin xyz="1 0 0" rpy="3.14159 0 0" />
    <axis xyz="0 0 1" />
  </joint>
</robot>
